### ISSUE TITLE
CloudQueue.CreateIfNotExistsAsync Hangs indefinitely 

### ISSUE THREAD

**Issue by Vandersteen** [Azure/azure-storage-net#997]
### Which service(blob, file, queue, table) does this issue concern?
Queue (as far as I tested)

### Which version of the SDK was used?
11.1.5 & 11.1.6

### Which platform are you using? (ex: .NET Core 2.1)
.NET Core 2.2

### What problem was encountered?
Code hangs indefinitely

### How can we reproduce the problem in the simplest way?
Close the local emulator

Run following example code:

```
            var requestOptions = new QueueRequestOptions
            {
                RetryPolicy = new ExponentialRetry(TimeSpan.FromMilliseconds(50), 5)
            };

            var queue = _queueClient.GetQueueReference(queueName);

            logger.LogInformation("Creating queue if it does not exist");
            var created = await queue.CreateIfNotExistsAsync(requestOptions, new OperationContext());
            logger.LogInformation("Queue created: {0}", created);
```

I can see it retrying 5 times, and after that it just hangs indefinitely. (I waited over an hour)
In other words the log line: "Queue created" is never hit

### Have you found a mitigation/solution?
No I did not, I tried multiple retry policies and no retry policy.
I tried messing with the following properties  of the `QueueRequestOptions`

```
                MaximumExecutionTime = TimeSpan.FromSeconds(10),
                NetworkTimeout = TimeSpan.FromSeconds(10),
                ServerTimeout = TimeSpan.FromSeconds(10)
```

It had no effect.

I was also able to trigger this a few times on `queue.GetMessageAsync` but not consistently.

**Comment by firephantomassasin** [comment]
Try to update to 11.1.7 or to latest version which is 11.2.2, i also experienced this kind of issue and it looks like the issue is gone in 11.1.7 above

**Comment by jetuser** [comment]
I am seeing similar issues on Azure.Storage.Queues 12.14.0 as well
